libnavigation-base/src/test/java/com/mapbox/navigation/base/routerefresh/MapboxRouteRefreshAdapterTest.kt

```diff
@@ -0,0 +1,115 @@
+package com.mapbox.navigation.base.routerefresh
+
+import android.location.Location
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.internal.route.RouteUrl
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+
+class MapboxRouteRefreshAdapterTest {
+
+    private val accessToken = "pk.1234"
+
+    private lateinit var routeRefreshAdapter: MapboxRouteRefreshAdapter
+    private lateinit var location: Location
+
+    companion object {
+        private const val DEFAULT_REROUTE_BEARING_ANGLE = 11f
+    }
+
+    @Before
+    fun setup(){
+        routeRefreshAdapter = MapboxRouteRefreshAdapter()
+
+        mockLocation()
+    }
+
+    @Test
+    fun reRoute_called_with_null_bearings() {
+        val routeOptions = provideRouteOptionsWithCoordinates()
+        val routeProgress: RouteProgress = mockk(relaxed = true)
+
+        val newRouteOptions = routeRefreshAdapter.newRouteOptions(routeOptions, routeProgress, location)
+
+        val expectedBearings = listOf(
+            listOf(DEFAULT_REROUTE_BEARING_ANGLE.toDouble(), MapboxRouteRefreshAdapter.DEFAULT_REROUTE_BEARING_TOLERANCE),
+            null,
+            null,
+            null
+        )
+        val actualBearings = newRouteOptions.bearingsList()
+
+        Assert.assertEquals(expectedBearings, actualBearings)
+    }
+
+    @Test
+    fun reRoute_called_with_bearings() {
+        val routeOptions = provideRouteOptionsWithCoordinatesAndBearings()
+        val routeProgress: RouteProgress = mockk(relaxed = true)
+
+        val newRouteOptions = routeRefreshAdapter.newRouteOptions(routeOptions, routeProgress, location)
+
+        val expectedBearings = listOf(
+            listOf(DEFAULT_REROUTE_BEARING_ANGLE.toDouble(), 10.0),
+            listOf(20.0, 20.0),
+            listOf(30.0, 30.0),
+            listOf(40.0, 40.0)
+        )
+        val actualBearings = newRouteOptions.bearingsList()
+
+        Assert.assertEquals(expectedBearings, actualBearings)
+    }
+
```

Comment:
can we add the no-op cases for reroute? e.g., `reRoute not called with invalid route`

Reply:
this test must be a part of `MapboxNavigation`, but here we can test that `MapboxRouteRefreshAdapter` returns null on non-valide options

Reply:
looks like "no_options_on_invalid_input" was added, or i missed it. but that's what this comment was asking for, was to add that test 👍  💯

Reply:
i don't quite understand why the logic needs to be in MapboxNavigation though